core/src/main/java/org/infinispan/commands/write/BackupMultiKeyWriteRpcCommand.java

```diff
@@ -0,0 +1,397 @@
+package org.infinispan.commands.write;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import org.infinispan.commands.CommandInvocationId;
+import org.infinispan.commands.TopologyAffectedCommand;
+import org.infinispan.commands.functional.ReadWriteManyCommand;
+import org.infinispan.commands.functional.ReadWriteManyEntriesCommand;
+import org.infinispan.commands.functional.WriteOnlyManyCommand;
+import org.infinispan.commands.functional.WriteOnlyManyEntriesCommand;
+import org.infinispan.commands.remote.BaseRpcCommand;
+import org.infinispan.commons.marshall.MarshallUtil;
+import org.infinispan.commons.util.EnumUtil;
+import org.infinispan.context.Flag;
+import org.infinispan.context.InvocationContext;
+import org.infinispan.context.InvocationContextFactory;
+import org.infinispan.context.impl.FlagBitSets;
+import org.infinispan.distribution.ch.KeyPartitioner;
+import org.infinispan.encoding.DataConversion;
+import org.infinispan.factories.ComponentRegistry;
+import org.infinispan.functional.impl.Params;
+import org.infinispan.interceptors.AsyncInterceptorChain;
+import org.infinispan.metadata.Metadata;
+import org.infinispan.notifications.cachelistener.CacheNotifier;
+import org.infinispan.util.ByteString;
+
+/**
+ * A command sent from the primary owner to the backup owners for a {@link PutMapCommand}.
+ * <p>
+ * This command is only visited by the backups owner and in a remote context. The command order is set by {@code
+ * segmentsAndSequences} map.
+ *
+ * @author Pedro Ruivo
+ * @since 9.0
+ */
+public class BackupMultiKeyWriteRpcCommand extends BaseRpcCommand implements TopologyAffectedCommand {
```

Comment:
I tried to merge BackupWriteRpcCommand and BackupMultiKeyWriteRpcCommand, but it was kind of ugly... What you have in mind? 
I guess the best approach would be to create a command for each write command.

Reply:
My suggestion was not to merge those two but the opposite: split BackupMultiKey* for commands that use a map and set for entries/keys. - I'd prefere. Also you could just use `Object` for `consumer/biConsumer/function/biFunction` as the type is carried in `Operation` anyway, you can cast it.

What modifications have you done so that this comment got outdated?